users/views.py

```diff
@@ -1,3 +1,68 @@
-from django.shortcuts import render
+import json, bcrypt, jwt
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+from django.core.exceptions import ValidationError
+
+from users.models     import User
+from users.validators import validate_email, validate_password
+from my_settings      import SECRET_KEY, ALGORITHM
+
+class SignUpView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            username        = data.get('username')
+            email           = data.get('email')
+            password        = data.get('password')
+            phone_number    = data.get('phone_number')
+            address         = data.get('address')
+            hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+
+            if validate_email(email)==None:
+                return JsonResponse({'messasge':'INVALID EMAIL'}, status=400) 
+            if validate_password(password)==None:
+                return JsonResponse({'messasge':'INVALID PASSWORD'}, status=400) 
+            if User.objects.filter(email = email).exists():
+                return JsonResponse({'messasge':'E-MAIL ALREADY EXISTED'}, status=400)
```

Comment:
- if문 사이에 공백 추가해 주세요.
- if문에서 ==None 이라는 표현은 잘 안 사용합니다. None이 False을 의미하기 때문에 not을 이용해서 구현할 수 있습니다.
```suggestion
            if not validate_email(email):
                return JsonResponse({'messasge':'INVALID EMAIL'}, status=400) 
            
            if not validate_password(password):
                return JsonResponse({'messasge':'INVALID PASSWORD'}, status=400) 
            
            if User.objects.filter(email = email).exists():
                return JsonResponse({'messasge':'E-MAIL ALREADY EXISTED'}, status=400)
```

Reply:
바로 수정하겠습니다!